CONTRIBUTING.md

```diff
@@ -174,6 +174,93 @@ if desired.
   Changelog: None
   ```
 
+### Structuring your commits
+More often than not, your pull request will come as a set of commits, not just
+a single one. This is especially recommended in case of larger changesets.
+
+In that case, please make sure that each commit constitutes a cohesive, logical
+whole, e.g.  modifies a given package, function, or application layer. There are many ways
+to conceptually divide your changeset, depending on its size and content - this is up to you.
+It's just important that unrelated changes are not mixed up together in unrelated commits.
+
+This is to ensure that:
+* your PR is easy to browse and review
+* git log is easier to digest
+
+#### Example
+
+* Bad:
+
+```
+Refactor X.
+
+Changelog: None
+```
```

Comment:
Would prefer if the examples where not as abstract.

And the examples are not following the guidelines, as they are missing sign-off lines :)

Reply:
I would at least vote for doing one or the other: Either include both changelog and signoff, or omit both, since in this context I think the changelog is also noise. I'm leaning slightly towards including both (people might only glance at one or two examples), but omitting them is fine by me if you want less noise.

I didn't really react to the abstractness. I think it's a pretty understandable workflow.

Reply:
> as for specific examples, I can provide a backend related one - and this is the client's repo - don't want to confuse folks. could you procure a client-related example?

I would not go that far, to actually point out existing commits and provide them as examples and having separate for client and backend. I believe that we can share them.